python/rmm/_lib/device_buffer.pyx

```diff
@@ -133,6 +133,27 @@ cdef class DeviceBuffer:
         }
         return intf
 
+    def copy(self):
+        """Returns a deep copy of DeviceBuffer
+
+        Returns
+        -------
+        A deep copy of existing ``DeviceBuffer``
+
+        Examples
+        --------
+        >>> import rmm
+        >>> db = rmm.DeviceBuffer.to_device(b"abc")
+        >>> db_copy = db.copy()
+        >>> db.copy_to_host()
+        array([97, 98, 99], dtype=uint8)
+        >>> db_copy.copy_to_host()
+        array([97, 98, 99], dtype=uint8)
+        """
+        new_copy = DeviceBuffer(size=self.size)
+        copy_device_to_ptr(self.ptr, new_copy.ptr, self.size)
```

Comment:
The underlying C++ class has a deep copy constructor. Isn't there some way of just calling that?

Reply:
Could you link me to the constructor Mark?

Reply:
https://github.com/rapidsai/rmm/blob/8f7a432d365c19ba2f143b5163ebc83d62fa6674/include/rmm/device_buffer.hpp#L55-L56

https://github.com/rapidsai/rmm/blob/8f7a432d365c19ba2f143b5163ebc83d62fa6674/include/rmm/device_buffer.hpp#L140-L166

Reply:
I think yes, it would be possible to leverage the C++ copy ctor. Something like this:

```python
def copy(self):
    ret = DeviceBuffer.__new__(DeviceBuffer)
    cdef unique_ptr[device_buffer] c_obj = make_unique[device_buffer](self.c_obj)  # should invoke copy ctor?
    ret.c_obj = ret
    ret.mr = self.mr
    ret.stream = self.stream
    return ret
```

Reply:
Updated to use the existing `ctor`